subsys/fs/shell.c

```diff
@@ -334,6 +334,66 @@ static int cmd_read(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_dump(const struct shell *shell, size_t argc, char **argv)
+{
+	char path[MAX_PATH_LEN];
+	struct fs_dirent dirent;
+	struct fs_file_t file;
+	int count;
+	int err;
+
+	create_abs_path(argv[1], path, sizeof(path));
+
+	if (argc > 2) {
+		count = strtol(argv[2], NULL, 0);
+		if (count <= 0) {
+			count = INT_MAX;
+		}
+	} else {
+		count = INT_MAX;
+	}
+
+	err = fs_stat(path, &dirent);
+	if (err) {
+		shell_error(shell, "Failed to obtain file %s (err: %d)",
+			    path, err);
+		return -ENOEXEC;
+	}
+
+	if (dirent.type != FS_DIR_ENTRY_FILE) {
+		shell_error(shell, "Note a file %s", path);
+		return -ENOEXEC;
+	}
+
+	fs_file_t_init(&file);
+	err = fs_open(&file, path, FS_O_CREATE | FS_O_RDWR);
```

Comment:
You do not need RDWR to read-only operations, you probably do not want to create file when it does not exist, which `FS_O_CREATE` will do.
The `FS_O_READ` flag suffice here. 
I know it has been done the same way in `cmd_read`, just fixed that.